Subject: [PATCH] PLANNER-2911 Remove hibernate score types
---
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendable/BendableScoreHibernateType.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendable/BendableScoreHibernateType.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendable/BendableScoreHibernateType.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendable/BendableScoreHibernateType.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,31 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.bendable;
-
-import java.util.Properties;
-
-import org.hibernate.type.StandardBasicTypes;
-import org.hibernate.usertype.CompositeUserType;
-import org.hibernate.usertype.ParameterizedType;
-import org.optaplanner.core.impl.score.buildin.BendableScoreDefinition;
-import org.optaplanner.persistence.jpa.impl.score.AbstractScoreHibernateType;
-
-/**
- * @deprecated This class has been deprecated as the Hibernate 6 does not provide full backward compatibility
- * for the {@link CompositeUserType}. The class will remain available in the OptaPlanner 8 releases to provide
- * integration with Hibernate 5 but will be removed in the OptaPlanner 9.37.0.Beta release.
- * To integrate the {@link org.optaplanner.core.api.score.buildin.bendable.BendableScore} with Hibernate 6,
- * either use the score converter
- * {@link org.optaplanner.persistence.jpa.api.score.buildin.bendable.BendableScoreConverter})
- * or implement the {@link CompositeUserType} yourself.
- */
-@Deprecated(forRemoval = true)
-public class BendableScoreHibernateType extends AbstractScoreHibernateType implements ParameterizedType {
-
-    @Override
-    public void setParameterValues(Properties parameterMap) {
-        int hardLevelsSize = extractIntParameter(parameterMap, "hardLevelsSize");
-        int softLevelsSize = extractIntParameter(parameterMap, "softLevelsSize");
-        scoreDefinition = new BendableScoreDefinition(hardLevelsSize, softLevelsSize);
-        type = StandardBasicTypes.INTEGER;
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendablebigdecimal/BendableBigDecimalScoreHibernateType.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendablebigdecimal/BendableBigDecimalScoreHibernateType.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendablebigdecimal/BendableBigDecimalScoreHibernateType.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendablebigdecimal/BendableBigDecimalScoreHibernateType.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,31 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.bendablebigdecimal;
-
-import java.util.Properties;
-
-import org.hibernate.type.StandardBasicTypes;
-import org.hibernate.usertype.CompositeUserType;
-import org.hibernate.usertype.ParameterizedType;
-import org.optaplanner.core.impl.score.buildin.BendableBigDecimalScoreDefinition;
-import org.optaplanner.persistence.jpa.impl.score.AbstractScoreHibernateType;
-
-/**
- * @deprecated This class has been deprecated as the Hibernate 6 does not provide full backward compatibility
- * for the {@link CompositeUserType}. The class will remain available in the OptaPlanner 8 releases to provide
- * integration with Hibernate 5 but will be removed in the OptaPlanner 9.37.0.Beta release.
- * To integrate the {@link org.optaplanner.core.api.score.buildin.bendablebigdecimal.BendableBigDecimalScore}
- * with Hibernate 6, either use the score converter
- * {@link org.optaplanner.persistence.jpa.api.score.buildin.bendablebigdecimal.BendableBigDecimalScoreConverter})
- * or implement the {@link CompositeUserType} yourself.
- */
-@Deprecated(forRemoval = true)
-public class BendableBigDecimalScoreHibernateType extends AbstractScoreHibernateType implements ParameterizedType {
-
-    @Override
-    public void setParameterValues(Properties parameterMap) {
-        int hardLevelsSize = extractIntParameter(parameterMap, "hardLevelsSize");
-        int softLevelsSize = extractIntParameter(parameterMap, "softLevelsSize");
-        scoreDefinition = new BendableBigDecimalScoreDefinition(hardLevelsSize, softLevelsSize);
-        type = StandardBasicTypes.BIG_DECIMAL;
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendablelong/BendableLongScoreHibernateType.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendablelong/BendableLongScoreHibernateType.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendablelong/BendableLongScoreHibernateType.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendablelong/BendableLongScoreHibernateType.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,30 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.bendablelong;
-
-import java.util.Properties;
-
-import org.hibernate.type.StandardBasicTypes;
-import org.hibernate.usertype.CompositeUserType;
-import org.hibernate.usertype.ParameterizedType;
-import org.optaplanner.core.impl.score.buildin.BendableLongScoreDefinition;
-import org.optaplanner.persistence.jpa.impl.score.AbstractScoreHibernateType;
-
-/**
- * @deprecated This class has been deprecated as the Hibernate 6 does not provide full backward compatibility
- * for the {@link CompositeUserType}. The class will remain available in the OptaPlanner 8 releases to provide
- * integration with Hibernate 5 but will be removed in the OptaPlanner 9.37.0.Beta release.
- * To integrate the {@link org.optaplanner.core.api.score.buildin.bendablelong.BendableLongScore}
- * with Hibernate 6, either use the score converter
- * {@link org.optaplanner.persistence.jpa.api.score.buildin.bendablelong.BendableLongScoreConverter})
- * or implement the {@link CompositeUserType} yourself.
- */
-public class BendableLongScoreHibernateType extends AbstractScoreHibernateType implements ParameterizedType {
-
-    @Override
-    public void setParameterValues(Properties parameterMap) {
-        int hardLevelsSize = extractIntParameter(parameterMap, "hardLevelsSize");
-        int softLevelsSize = extractIntParameter(parameterMap, "softLevelsSize");
-        scoreDefinition = new BendableLongScoreDefinition(hardLevelsSize, softLevelsSize);
-        type = StandardBasicTypes.LONG;
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoft/HardMediumSoftScoreHibernateType.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoft/HardMediumSoftScoreHibernateType.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoft/HardMediumSoftScoreHibernateType.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoft/HardMediumSoftScoreHibernateType.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,25 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.hardmediumsoft;
-
-import org.hibernate.type.StandardBasicTypes;
-import org.hibernate.usertype.CompositeUserType;
-import org.optaplanner.core.impl.score.buildin.HardMediumSoftScoreDefinition;
-import org.optaplanner.persistence.jpa.impl.score.AbstractScoreHibernateType;
-
-/**
- * @deprecated This class has been deprecated as the Hibernate 6 does not provide full backward compatibility
- * for the {@link CompositeUserType}. The class will remain available in the OptaPlanner 8 releases to provide
- * integration with Hibernate 5 but will be removed in the OptaPlanner 9.37.0.Beta release.
- * To integrate the {@link org.optaplanner.core.api.score.buildin.hardmediumsoft.HardMediumSoftScore}
- * with Hibernate 6, either use the score converter
- * {@link org.optaplanner.persistence.jpa.api.score.buildin.hardmediumsoft.HardMediumSoftScoreConverter})
- * or implement the {@link CompositeUserType} yourself.
- */
-@Deprecated(forRemoval = true)
-public class HardMediumSoftScoreHibernateType extends AbstractScoreHibernateType {
-
-    public HardMediumSoftScoreHibernateType() {
-        scoreDefinition = new HardMediumSoftScoreDefinition();
-        type = StandardBasicTypes.INTEGER;
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoftbigdecimal/HardMediumSoftBigDecimalScoreHibernateType.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoftbigdecimal/HardMediumSoftBigDecimalScoreHibernateType.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoftbigdecimal/HardMediumSoftBigDecimalScoreHibernateType.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoftbigdecimal/HardMediumSoftBigDecimalScoreHibernateType.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,25 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.hardmediumsoftbigdecimal;
-
-import org.hibernate.type.StandardBasicTypes;
-import org.hibernate.usertype.CompositeUserType;
-import org.optaplanner.core.impl.score.buildin.HardMediumSoftBigDecimalScoreDefinition;
-import org.optaplanner.persistence.jpa.impl.score.AbstractScoreHibernateType;
-
-/**
- * @deprecated This class has been deprecated as the Hibernate 6 does not provide full backward compatibility
- * for the {@link CompositeUserType}. The class will remain available in the OptaPlanner 8 releases to provide
- * integration with Hibernate 5 but will be removed in the OptaPlanner 9.37.0.Beta release.
- * To integrate the {@link org.optaplanner.core.api.score.buildin.hardmediumsoftbigdecimal.HardMediumSoftBigDecimalScore}
- * with Hibernate 6, either use the score converter
- * {@link org.optaplanner.persistence.jpa.api.score.buildin.hardmediumsoftbigdecimal.HardMediumSoftBigDecimalScoreConverter})
- * or implement the {@link CompositeUserType} yourself.
- */
-@Deprecated(forRemoval = true)
-public class HardMediumSoftBigDecimalScoreHibernateType extends AbstractScoreHibernateType {
-
-    public HardMediumSoftBigDecimalScoreHibernateType() {
-        scoreDefinition = new HardMediumSoftBigDecimalScoreDefinition();
-        type = StandardBasicTypes.BIG_DECIMAL;
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoftlong/HardMediumSoftLongScoreHibernateType.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoftlong/HardMediumSoftLongScoreHibernateType.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoftlong/HardMediumSoftLongScoreHibernateType.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoftlong/HardMediumSoftLongScoreHibernateType.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,25 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.hardmediumsoftlong;
-
-import org.hibernate.type.StandardBasicTypes;
-import org.hibernate.usertype.CompositeUserType;
-import org.optaplanner.core.impl.score.buildin.HardMediumSoftLongScoreDefinition;
-import org.optaplanner.persistence.jpa.impl.score.AbstractScoreHibernateType;
-
-/**
- * @deprecated This class has been deprecated as the Hibernate 6 does not provide full backward compatibility
- * for the {@link CompositeUserType}. The class will remain available in the OptaPlanner 8 releases to provide
- * integration with Hibernate 5 but will be removed in the OptaPlanner 9.37.0.Beta release.
- * To integrate the {@link org.optaplanner.core.api.score.buildin.hardmediumsoftlong.HardMediumSoftLongScore}
- * with Hibernate 6, either use the score converter
- * {@link org.optaplanner.persistence.jpa.api.score.buildin.hardmediumsoftlong.HardMediumSoftLongScoreConverter})
- * or implement the {@link CompositeUserType} yourself.
- */
-@Deprecated(forRemoval = true)
-public class HardMediumSoftLongScoreHibernateType extends AbstractScoreHibernateType {
-
-    public HardMediumSoftLongScoreHibernateType() {
-        scoreDefinition = new HardMediumSoftLongScoreDefinition();
-        type = StandardBasicTypes.LONG;
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoft/HardSoftScoreHibernateType.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoft/HardSoftScoreHibernateType.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoft/HardSoftScoreHibernateType.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoft/HardSoftScoreHibernateType.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,25 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.hardsoft;
-
-import org.hibernate.type.StandardBasicTypes;
-import org.hibernate.usertype.CompositeUserType;
-import org.optaplanner.core.impl.score.buildin.HardSoftScoreDefinition;
-import org.optaplanner.persistence.jpa.impl.score.AbstractScoreHibernateType;
-
-/**
- * @deprecated This class has been deprecated as the Hibernate 6 does not provide full backward compatibility
- * for the {@link CompositeUserType}. The class will remain available in the OptaPlanner 8 releases to provide
- * integration with Hibernate 5 but will be removed in the OptaPlanner 9.37.0.Beta release.
- * To integrate the {@link org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScore}
- * with Hibernate 6, either use the score converter
- * {@link org.optaplanner.persistence.jpa.api.score.buildin.hardsoft.HardSoftScoreConverter})
- * or implement the {@link CompositeUserType} yourself.
- */
-@Deprecated(forRemoval = true)
-public class HardSoftScoreHibernateType extends AbstractScoreHibernateType {
-
-    public HardSoftScoreHibernateType() {
-        scoreDefinition = new HardSoftScoreDefinition();
-        type = StandardBasicTypes.INTEGER;
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoftbigdecimal/HardSoftBigDecimalScoreHibernateType.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoftbigdecimal/HardSoftBigDecimalScoreHibernateType.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoftbigdecimal/HardSoftBigDecimalScoreHibernateType.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoftbigdecimal/HardSoftBigDecimalScoreHibernateType.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,25 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.hardsoftbigdecimal;
-
-import org.hibernate.type.StandardBasicTypes;
-import org.hibernate.usertype.CompositeUserType;
-import org.optaplanner.core.impl.score.buildin.HardSoftBigDecimalScoreDefinition;
-import org.optaplanner.persistence.jpa.impl.score.AbstractScoreHibernateType;
-
-/**
- * @deprecated This class has been deprecated as the Hibernate 6 does not provide full backward compatibility
- * for the {@link CompositeUserType}. The class will remain available in the OptaPlanner 8 releases to provide
- * integration with Hibernate 5 but will be removed in the OptaPlanner 9.37.0.Beta release.
- * To integrate the {@link org.optaplanner.core.api.score.buildin.hardsoftbigdecimal.HardSoftBigDecimalScore}
- * with Hibernate 6, either use the score converter
- * {@link org.optaplanner.persistence.jpa.api.score.buildin.hardsoftbigdecimal.HardSoftBigDecimalScoreConverter})
- * or implement the {@link CompositeUserType} yourself.
- */
-@Deprecated(forRemoval = true)
-public class HardSoftBigDecimalScoreHibernateType extends AbstractScoreHibernateType {
-
-    public HardSoftBigDecimalScoreHibernateType() {
-        scoreDefinition = new HardSoftBigDecimalScoreDefinition();
-        type = StandardBasicTypes.BIG_DECIMAL;
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoftlong/HardSoftLongScoreHibernateType.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoftlong/HardSoftLongScoreHibernateType.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoftlong/HardSoftLongScoreHibernateType.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoftlong/HardSoftLongScoreHibernateType.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,25 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.hardsoftlong;
-
-import org.hibernate.type.StandardBasicTypes;
-import org.hibernate.usertype.CompositeUserType;
-import org.optaplanner.core.impl.score.buildin.HardSoftLongScoreDefinition;
-import org.optaplanner.persistence.jpa.impl.score.AbstractScoreHibernateType;
-
-/**
- * @deprecated This class has been deprecated as the Hibernate 6 does not provide full backward compatibility
- * for the {@link CompositeUserType}. The class will remain available in the OptaPlanner 8 releases to provide
- * integration with Hibernate 5 but will be removed in the OptaPlanner 9.37.0.Beta release.
- * To integrate the {@link org.optaplanner.core.api.score.buildin.hardsoftlong.HardSoftLongScore}
- * with Hibernate 6, either use the score converter
- * {@link org.optaplanner.persistence.jpa.api.score.buildin.hardsoftlong.HardSoftLongScoreConverter})
- * or implement the {@link CompositeUserType} yourself.
- */
-@Deprecated(forRemoval = true)
-public class HardSoftLongScoreHibernateType extends AbstractScoreHibernateType {
-
-    public HardSoftLongScoreHibernateType() {
-        scoreDefinition = new HardSoftLongScoreDefinition();
-        type = StandardBasicTypes.LONG;
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/simple/SimpleScoreHibernateType.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/simple/SimpleScoreHibernateType.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/simple/SimpleScoreHibernateType.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/simple/SimpleScoreHibernateType.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,25 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.simple;
-
-import org.hibernate.type.StandardBasicTypes;
-import org.hibernate.usertype.CompositeUserType;
-import org.optaplanner.core.impl.score.buildin.SimpleScoreDefinition;
-import org.optaplanner.persistence.jpa.impl.score.AbstractScoreHibernateType;
-
-/**
- * @deprecated This class has been deprecated as the Hibernate 6 does not provide full backward compatibility
- * for the {@link CompositeUserType}. The class will remain available in the OptaPlanner 8 releases to provide
- * integration with Hibernate 5 but will be removed in the OptaPlanner 9.37.0.Beta release.
- * To integrate the {@link org.optaplanner.core.api.score.buildin.simple.SimpleScore}
- * with Hibernate 6, either use the score converter
- * {@link org.optaplanner.persistence.jpa.api.score.buildin.simple.SimpleScoreConverter})
- * or implement the {@link CompositeUserType} yourself.
- */
-@Deprecated(forRemoval = true)
-public class SimpleScoreHibernateType extends AbstractScoreHibernateType {
-
-    public SimpleScoreHibernateType() {
-        scoreDefinition = new SimpleScoreDefinition();
-        type = StandardBasicTypes.INTEGER;
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/simplebigdecimal/SimpleBigDecimalScoreHibernateType.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/simplebigdecimal/SimpleBigDecimalScoreHibernateType.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/simplebigdecimal/SimpleBigDecimalScoreHibernateType.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/simplebigdecimal/SimpleBigDecimalScoreHibernateType.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,25 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.simplebigdecimal;
-
-import org.hibernate.type.StandardBasicTypes;
-import org.hibernate.usertype.CompositeUserType;
-import org.optaplanner.core.impl.score.buildin.SimpleBigDecimalScoreDefinition;
-import org.optaplanner.persistence.jpa.impl.score.AbstractScoreHibernateType;
-
-/**
- * @deprecated This class has been deprecated as the Hibernate 6 does not provide full backward compatibility
- * for the {@link CompositeUserType}. The class will remain available in the OptaPlanner 8 releases to provide
- * integration with Hibernate 5 but will be removed in the OptaPlanner 9.37.0.Beta release.
- * To integrate the {@link org.optaplanner.core.api.score.buildin.simplebigdecimal.SimpleBigDecimalScore}
- * with Hibernate 6, either use the score converter
- * {@link org.optaplanner.persistence.jpa.api.score.buildin.simplebigdecimal.SimpleBigDecimalScoreConverter})
- * or implement the {@link CompositeUserType} yourself.
- */
-@Deprecated(forRemoval = true)
-public class SimpleBigDecimalScoreHibernateType extends AbstractScoreHibernateType {
-
-    public SimpleBigDecimalScoreHibernateType() {
-        scoreDefinition = new SimpleBigDecimalScoreDefinition();
-        type = StandardBasicTypes.BIG_DECIMAL;
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/simplelong/SimpleLongScoreHibernateType.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/simplelong/SimpleLongScoreHibernateType.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/simplelong/SimpleLongScoreHibernateType.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/buildin/simplelong/SimpleLongScoreHibernateType.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,25 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.simplelong;
-
-import org.hibernate.type.StandardBasicTypes;
-import org.hibernate.usertype.CompositeUserType;
-import org.optaplanner.core.impl.score.buildin.SimpleLongScoreDefinition;
-import org.optaplanner.persistence.jpa.impl.score.AbstractScoreHibernateType;
-
-/**
- * @deprecated This class has been deprecated as the Hibernate 6 does not provide full backward compatibility
- * for the {@link CompositeUserType}. The class will remain available in the OptaPlanner 8 releases to provide
- * integration with Hibernate 5 but will be removed in the OptaPlanner 9.37.0.Beta release.
- * To integrate the {@link org.optaplanner.core.api.score.buildin.simplelong.SimpleLongScore}
- * with Hibernate 6, either use the score converter
- * {@link org.optaplanner.persistence.jpa.api.score.buildin.simplelong.SimpleLongScoreConverter})
- * or implement the {@link CompositeUserType} yourself.
- */
-@Deprecated(forRemoval = true)
-public class SimpleLongScoreHibernateType extends AbstractScoreHibernateType {
-
-    public SimpleLongScoreHibernateType() {
-        scoreDefinition = new SimpleLongScoreDefinition();
-        type = StandardBasicTypes.LONG;
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/AbstractScoreHibernateType.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/AbstractScoreHibernateType.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/AbstractScoreHibernateType.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/main/java/org/optaplanner/persistence/jpa/impl/score/AbstractScoreHibernateType.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,228 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score;
-
-import java.io.Serializable;
-import java.math.BigDecimal;
-import java.sql.PreparedStatement;
-import java.sql.ResultSet;
-import java.sql.SQLException;
-import java.util.Arrays;
-import java.util.Properties;
-
-import org.hibernate.engine.spi.SharedSessionContractImplementor;
-import org.hibernate.type.AbstractSingleColumnStandardBasicType;
-import org.hibernate.type.BigDecimalType;
-import org.hibernate.type.DoubleType;
-import org.hibernate.type.IntegerType;
-import org.hibernate.type.LongType;
-import org.hibernate.type.StandardBasicTypes;
-import org.hibernate.type.Type;
-import org.hibernate.usertype.CompositeUserType;
-import org.optaplanner.core.api.score.Score;
-import org.optaplanner.core.impl.score.definition.ScoreDefinition;
-
-/**
- * This class is Hibernate specific, because JPA 2.1's @Converter currently
- * cannot handle 1 class mapping to multiple SQL columns.
- *
- * @deprecated This class has been deprecated as the Hibernate 6 does not provide full backward compatibility
- * for the {@link CompositeUserType}. The class will remain available in the OptaPlanner 8 releases to provide
- * integration with Hibernate 5 but will be removed in the OptaPlanner 9.37.0.Beta release.
- * To integrate the {@link org.optaplanner.core.api.domain.solution.PlanningScore} with Hibernate 6,
- * either use the score converters
- * {@link org.optaplanner.persistence.jpa.api.score.buildin}) or implement the {@link CompositeUserType} yourself.
- */
-@Deprecated(forRemoval = true)
-public abstract class AbstractScoreHibernateType implements CompositeUserType {
-
-    protected ScoreDefinition scoreDefinition;
-    protected AbstractSingleColumnStandardBasicType type;
-
-    protected int extractIntParameter(Properties parameterMap, String parameterName) {
-        String valueString = (String) parameterMap.get(parameterName);
-        if (valueString == null) {
-            throw new IllegalArgumentException("The parameter " + parameterName + " (" + valueString
-                    + ") is missing for " + getClass().getSimpleName() + ".");
-        }
-        try {
-            return Integer.parseInt(valueString);
-        } catch (NumberFormatException e) {
-            throw new IllegalArgumentException("The parameter " + parameterName + " (" + valueString
-                    + ") is not a valid integer for " + getClass().getSimpleName() + ".");
-        }
-    }
-
-    @Override
-    public Class returnedClass() {
-        return scoreDefinition.getScoreClass();
-    }
-
-    @Override
-    public String[] getPropertyNames() {
-        int levelsSize = scoreDefinition.getLevelsSize();
-        String[] propertyNames = new String[levelsSize + 1];
-        propertyNames[0] = scoreDefinition.getInitLabel();
-        String[] levelLabels = scoreDefinition.getLevelLabels();
-        for (int i = 0; i < levelsSize; i++) {
-            String propertyName = levelLabels[i].replaceAll("\\s+s", "S").replaceAll("\\s+(\\d)", "$1");
-            if (!propertyName.matches("[\\w\\d]+")) {
-                throw new IllegalStateException("The levelLabel (" + levelLabels[i] + ") is not yet supported.");
-            }
-            propertyNames[i + 1] = propertyName;
-        }
-        return propertyNames;
-    }
-
-    @Override
-    public Type[] getPropertyTypes() {
-        int levelsSize = scoreDefinition.getLevelsSize();
-        Type[] propertyTypes = new Type[levelsSize + 1];
-        propertyTypes[0] = StandardBasicTypes.INTEGER;
-        for (int i = 0; i < levelsSize; i++) {
-            propertyTypes[i + 1] = type;
-        }
-        return propertyTypes;
-    }
-
-    @Override
-    public Object getPropertyValue(Object o, int propertyIndex) {
-        if (o == null) {
-            return null;
-        }
-        Score score = (Score) o;
-        if (propertyIndex == 0) {
-            return score.initScore();
-        }
-        if (propertyIndex >= scoreDefinition.getLevelsSize() + 1) {
-            throw new IllegalArgumentException("The propertyIndex (" + propertyIndex
-                    + ") must be lower than the levelsSize for score (" + score + ").");
-        }
-        Number[] levelNumbers = score.toLevelNumbers();
-        return levelNumbers[propertyIndex - 1];
-    }
-
-    @Override
-    public Score nullSafeGet(ResultSet resultSet, String[] names, SharedSessionContractImplementor session, Object owner)
-            throws SQLException {
-        if (resultSet == null) {
-            return null;
-        }
-        int nullCount = 0;
-        Integer initScore = (Integer) StandardBasicTypes.INTEGER.nullSafeGet(resultSet, names[0], session, owner);
-        if (initScore == null) {
-            nullCount++;
-        }
-        int levelsSize = scoreDefinition.getLevelsSize();
-        Number[] levelNumbers = new Number[levelsSize];
-        for (int i = 0; i < levelsSize; i++) {
-            Number levelNumber = (Number) type.nullSafeGet(resultSet, names[i + 1], session, owner);
-            if (levelNumber == null) {
-                nullCount++;
-            } else {
-                levelNumbers[i] = levelNumber;
-            }
-        }
-        if (nullCount == levelsSize + 1) {
-            return null;
-        } else if (nullCount != 0) {
-            throw new IllegalStateException("The nullCount (" + nullCount
-                    + ") must be 0 or levelsSize (" + levelsSize + ") for " + getClass().getSimpleName() + ".");
-        }
-        Score score = scoreDefinition.fromLevelNumbers(initScore, levelNumbers);
-        if (score == null) {
-            throw new IllegalStateException("The levelNumbers (" + Arrays.toString(levelNumbers)
-                    + ") must not build a score (" + null + ") that is null.");
-        }
-        return score;
-    }
-
-    @Override
-    public void nullSafeSet(PreparedStatement statement, Object value, int parameterIndex,
-            SharedSessionContractImplementor session)
-            throws SQLException {
-        int levelsSize = scoreDefinition.getLevelsSize();
-        if (value == null) {
-            statement.setNull(parameterIndex, StandardBasicTypes.INTEGER.sqlType()); // initScore
-            for (int i = 0; i < levelsSize; i++) {
-                statement.setNull(parameterIndex + i + 1, type.sqlType());
-            }
-            return;
-        }
-        Score score = (Score) value;
-        statement.setInt(parameterIndex, score.initScore());
-        Number[] levelNumbers = score.toLevelNumbers();
-        if (levelNumbers.length != levelsSize) {
-            throw new IllegalStateException("The levelNumbers length (" + levelNumbers.length + ") for score (" + score
-                    + ") must be equal to the levelsSize (" + levelsSize + ") for " + getClass().getSimpleName() + ".");
-        }
-        for (int i = 0; i < levelsSize; i++) {
-            if (type == IntegerType.INSTANCE) {
-                statement.setInt(parameterIndex + i + 1, (Integer) levelNumbers[i]);
-            } else if (type == LongType.INSTANCE) {
-                statement.setLong(parameterIndex + i + 1, (Long) levelNumbers[i]);
-            } else if (type == DoubleType.INSTANCE) {
-                statement.setDouble(parameterIndex + i + 1, (Double) levelNumbers[i]);
-            } else if (type == BigDecimalType.INSTANCE) {
-                statement.setBigDecimal(parameterIndex + i + 1, (BigDecimal) levelNumbers[i]);
-            } else {
-                throw new IllegalStateException("The type (" + type + ") is not yet supported.");
-            }
-        }
-    }
-
-    // ************************************************************************
-    // Mutable related methods
-    // ************************************************************************
-
-    @Override
-    public boolean isMutable() {
-        return false;
-    }
-
-    @Override
-    public Object deepCopy(Object value) {
-        return value; // Score is immutable
-    }
-
-    @Override
-    public Object replace(Object original, Object target, SharedSessionContractImplementor session, Object owner) {
-        return original; // Score is immutable
-    }
-
-    @Override
-    public void setPropertyValue(Object component, int property, Object value) {
-        throw new UnsupportedOperationException("A Score is immutable.");
-    }
-
-    // ************************************************************************
-    // Other methods
-    // ************************************************************************
-
-    @Override
-    public boolean equals(Object a, Object b) {
-        if (a == b) {
-            return true;
-        } else if (a == null || b == null) {
-            return false;
-        }
-        return a.equals(b);
-    }
-
-    @Override
-    public int hashCode(Object o) {
-        if (o == null) {
-            return 0;
-        }
-        return o.hashCode();
-    }
-
-    @Override
-    public Serializable disassemble(Object value, SharedSessionContractImplementor session) {
-        return (Serializable) value;
-    }
-
-    @Override
-    public Object assemble(Serializable cached, SharedSessionContractImplementor session, Object owner) {
-        return cached;
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendable/BendableScoreHibernateTypeTest.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendable/BendableScoreHibernateTypeTest.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendable/BendableScoreHibernateTypeTest.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendable/BendableScoreHibernateTypeTest.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,52 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.bendable;
-
-import javax.persistence.Column;
-import javax.persistence.Entity;
-
-import org.hibernate.annotations.Columns;
-import org.hibernate.annotations.Parameter;
-import org.hibernate.annotations.TypeDef;
-import org.junit.jupiter.api.Test;
-import org.optaplanner.core.api.score.buildin.bendable.BendableScore;
-import org.optaplanner.persistence.jpa.impl.AbstractScoreJpaTest;
-
-import io.quarkus.test.junit.QuarkusTest;
-
-@QuarkusTest
-class BendableScoreHibernateTypeTest extends AbstractScoreJpaTest {
-
-    @Test
-    void persistAndMerge() {
-        persistAndMerge(new BendableScoreHibernateTypeTestJpaEntity(BendableScore.zero(3, 2)),
-                BendableScore.of(new int[] { 10000, 2000, 300 }, new int[] { 40, 5 }),
-                BendableScore.ofUninitialized(-7, new int[] { 10000, 2000, 300 }, new int[] { 40, 5 }));
-    }
-
-    @Entity
-    @TypeDef(defaultForType = BendableScore.class, typeClass = BendableScoreHibernateType.class, parameters = {
-            @Parameter(name = "hardLevelsSize", value = "3"), @Parameter(name = "softLevelsSize", value = "2") })
-    static class BendableScoreHibernateTypeTestJpaEntity extends AbstractTestJpaEntity<BendableScore> {
-
-        @Columns(columns = { @Column(name = "initScore"),
-                @Column(name = "hard0Score"), @Column(name = "hard1Score"), @Column(name = "hard2Score"),
-                @Column(name = "soft0Score"), @Column(name = "soft1Score") })
-        protected BendableScore score;
-
-        BendableScoreHibernateTypeTestJpaEntity() {
-        }
-
-        public BendableScoreHibernateTypeTestJpaEntity(BendableScore score) {
-            this.score = score;
-        }
-
-        @Override
-        public BendableScore getScore() {
-            return score;
-        }
-
-        @Override
-        public void setScore(BendableScore score) {
-            this.score = score;
-        }
-    }
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendablebigdecimal/BendableBigDecimalScoreHibernateTypeTest.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendablebigdecimal/BendableBigDecimalScoreHibernateTypeTest.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendablebigdecimal/BendableBigDecimalScoreHibernateTypeTest.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendablebigdecimal/BendableBigDecimalScoreHibernateTypeTest.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,67 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.bendablebigdecimal;
-
-import java.math.BigDecimal;
-
-import javax.persistence.Column;
-import javax.persistence.Entity;
-
-import org.hibernate.annotations.Columns;
-import org.hibernate.annotations.Parameter;
-import org.hibernate.annotations.TypeDef;
-import org.junit.jupiter.api.Test;
-import org.optaplanner.core.api.score.buildin.bendablebigdecimal.BendableBigDecimalScore;
-import org.optaplanner.persistence.jpa.impl.AbstractScoreJpaTest;
-
-import io.quarkus.test.junit.QuarkusTest;
-
-@QuarkusTest
-class BendableBigDecimalScoreHibernateTypeTest extends AbstractScoreJpaTest {
-
-    @Test
-    void persistAndMerge() {
-        persistAndMerge(new BendableBigDecimalScoreHibernateTypeTestJpaEntity(BendableBigDecimalScore.zero(3, 2)),
-                BendableBigDecimalScore.of(
-                        new BigDecimal[] { new BigDecimal("10000.00001"), new BigDecimal("2000.00020"),
-                                new BigDecimal("300.00300") },
-                        new BigDecimal[] { new BigDecimal("40.04000"), new BigDecimal("5.50000") }),
-                BendableBigDecimalScore.ofUninitialized(-7,
-                        new BigDecimal[] { new BigDecimal("10000.00001"), new BigDecimal("2000.00020"),
-                                new BigDecimal("300.00300") },
-                        new BigDecimal[] { new BigDecimal("40.04000"), new BigDecimal("5.50000") }));
-    }
-
-    @Entity
-    @TypeDef(defaultForType = BendableBigDecimalScore.class, typeClass = BendableBigDecimalScoreHibernateType.class,
-            parameters = {
-                    @Parameter(name = "hardLevelsSize", value = "3"), @Parameter(name = "softLevelsSize", value = "2") })
-    static class BendableBigDecimalScoreHibernateTypeTestJpaEntity extends AbstractTestJpaEntity<BendableBigDecimalScore> {
-
-        @Columns(columns = {
-                @Column(name = "initScore"),
-                @Column(name = "hard0Score", precision = 10, scale = 5),
-                @Column(name = "hard1Score", precision = 10, scale = 5),
-                @Column(name = "hard2Score", precision = 10, scale = 5),
-                @Column(name = "soft0Score", precision = 10, scale = 5),
-                @Column(name = "soft1Score", precision = 10, scale = 5) })
-        protected BendableBigDecimalScore score;
-
-        BendableBigDecimalScoreHibernateTypeTestJpaEntity() {
-        }
-
-        public BendableBigDecimalScoreHibernateTypeTestJpaEntity(BendableBigDecimalScore score) {
-            this.score = score;
-        }
-
-        @Override
-        public BendableBigDecimalScore getScore() {
-            return score;
-        }
-
-        @Override
-        public void setScore(BendableBigDecimalScore score) {
-            this.score = score;
-        }
-
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendablelong/BendableLongScoreHibernateTypeTest.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendablelong/BendableLongScoreHibernateTypeTest.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendablelong/BendableLongScoreHibernateTypeTest.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/bendablelong/BendableLongScoreHibernateTypeTest.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,54 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.bendablelong;
-
-import javax.persistence.Column;
-import javax.persistence.Entity;
-
-import org.hibernate.annotations.Columns;
-import org.hibernate.annotations.Parameter;
-import org.hibernate.annotations.TypeDef;
-import org.junit.jupiter.api.Test;
-import org.optaplanner.core.api.score.buildin.bendablelong.BendableLongScore;
-import org.optaplanner.persistence.jpa.impl.AbstractScoreJpaTest;
-
-import io.quarkus.test.junit.QuarkusTest;
-
-@QuarkusTest
-class BendableLongScoreHibernateTypeTest extends AbstractScoreJpaTest {
-
-    @Test
-    void persistAndMerge() {
-        persistAndMerge(new BendableLongScoreHibernateTypeTestJpaEntity(BendableLongScore.zero(3, 2)),
-                BendableLongScore.of(new long[] { 10000L, 2000L, 300L }, new long[] { 40L, 5L }),
-                BendableLongScore.ofUninitialized(-7, new long[] { 10000L, 2000L, 300L }, new long[] { 40L, 5L }));
-    }
-
-    @Entity
-    @TypeDef(defaultForType = BendableLongScore.class, typeClass = BendableLongScoreHibernateType.class, parameters = {
-            @Parameter(name = "hardLevelsSize", value = "3"), @Parameter(name = "softLevelsSize", value = "2") })
-    static class BendableLongScoreHibernateTypeTestJpaEntity extends AbstractTestJpaEntity<BendableLongScore> {
-
-        @Columns(columns = { @Column(name = "initScore"),
-                @Column(name = "hard0Score"), @Column(name = "hard1Score"), @Column(name = "hard2Score"),
-                @Column(name = "soft0Score"), @Column(name = "soft1Score") })
-        protected BendableLongScore score;
-
-        BendableLongScoreHibernateTypeTestJpaEntity() {
-        }
-
-        public BendableLongScoreHibernateTypeTestJpaEntity(BendableLongScore score) {
-            this.score = score;
-        }
-
-        @Override
-        public BendableLongScore getScore() {
-            return score;
-        }
-
-        @Override
-        public void setScore(BendableLongScore score) {
-            this.score = score;
-        }
-
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoft/HardMediumSoftScoreHibernateTypeTest.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoft/HardMediumSoftScoreHibernateTypeTest.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoft/HardMediumSoftScoreHibernateTypeTest.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoft/HardMediumSoftScoreHibernateTypeTest.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,51 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.hardmediumsoft;
-
-import javax.persistence.Column;
-import javax.persistence.Entity;
-
-import org.hibernate.annotations.Columns;
-import org.hibernate.annotations.TypeDef;
-import org.junit.jupiter.api.Test;
-import org.optaplanner.core.api.score.buildin.hardmediumsoft.HardMediumSoftScore;
-import org.optaplanner.persistence.jpa.impl.AbstractScoreJpaTest;
-
-import io.quarkus.test.junit.QuarkusTest;
-
-@QuarkusTest
-class HardMediumSoftScoreHibernateTypeTest extends AbstractScoreJpaTest {
-
-    @Test
-    void persistAndMerge() {
-        persistAndMerge(new HardMediumSoftScoreHibernateTypeTestJpaEntity(HardMediumSoftScore.ZERO),
-                HardMediumSoftScore.of(-100, -20, -3),
-                HardMediumSoftScore.ofUninitialized(-7, -100, -20, -3));
-    }
-
-    @Entity
-    @TypeDef(defaultForType = HardMediumSoftScore.class, typeClass = HardMediumSoftScoreHibernateType.class)
-    static class HardMediumSoftScoreHibernateTypeTestJpaEntity extends AbstractTestJpaEntity<HardMediumSoftScore> {
-
-        @Columns(columns = { @Column(name = "initScore"),
-                @Column(name = "hardScore"), @Column(name = "mediumScore"), @Column(name = "softScore") })
-        protected HardMediumSoftScore score;
-
-        HardMediumSoftScoreHibernateTypeTestJpaEntity() {
-        }
-
-        public HardMediumSoftScoreHibernateTypeTestJpaEntity(HardMediumSoftScore score) {
-            this.score = score;
-        }
-
-        @Override
-        public HardMediumSoftScore getScore() {
-            return score;
-        }
-
-        @Override
-        public void setScore(HardMediumSoftScore score) {
-            this.score = score;
-        }
-
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoftbigdecimal/HardMediumSoftBigDecimalScoreHibernateTypeTest.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoftbigdecimal/HardMediumSoftBigDecimalScoreHibernateTypeTest.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoftbigdecimal/HardMediumSoftBigDecimalScoreHibernateTypeTest.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoftbigdecimal/HardMediumSoftBigDecimalScoreHibernateTypeTest.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,59 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.hardmediumsoftbigdecimal;
-
-import java.math.BigDecimal;
-
-import javax.persistence.Column;
-import javax.persistence.Entity;
-
-import org.hibernate.annotations.Columns;
-import org.hibernate.annotations.TypeDef;
-import org.junit.jupiter.api.Test;
-import org.optaplanner.core.api.score.buildin.hardmediumsoftbigdecimal.HardMediumSoftBigDecimalScore;
-import org.optaplanner.persistence.jpa.impl.AbstractScoreJpaTest;
-
-import io.quarkus.test.junit.QuarkusTest;
-
-@QuarkusTest
-class HardMediumSoftBigDecimalScoreHibernateTypeTest extends AbstractScoreJpaTest {
-
-    @Test
-    void persistAndMerge() {
-        persistAndMerge(new HardMediumSoftBigDecimalScoreHibernateTypeTestJpaEntity(HardMediumSoftBigDecimalScore.ZERO),
-                HardMediumSoftBigDecimalScore.of(new BigDecimal("-10.01000"), new BigDecimal("-4.32100"),
-                        new BigDecimal("-2.20000")),
-                HardMediumSoftBigDecimalScore.ofUninitialized(-7, new BigDecimal("-10.01000"), new BigDecimal("-4.32100"),
-                        new BigDecimal("-2.20000")));
-    }
-
-    @Entity
-    @TypeDef(defaultForType = HardMediumSoftBigDecimalScore.class, typeClass = HardMediumSoftBigDecimalScoreHibernateType.class)
-    static class HardMediumSoftBigDecimalScoreHibernateTypeTestJpaEntity
-            extends AbstractTestJpaEntity<HardMediumSoftBigDecimalScore> {
-
-        @Columns(columns = {
-                @Column(name = "initScore"),
-                @Column(name = "hardScore", precision = 10, scale = 5),
-                @Column(name = "mediumScore", precision = 10, scale = 5),
-                @Column(name = "softScore", precision = 10, scale = 5) })
-        protected HardMediumSoftBigDecimalScore score;
-
-        HardMediumSoftBigDecimalScoreHibernateTypeTestJpaEntity() {
-        }
-
-        public HardMediumSoftBigDecimalScoreHibernateTypeTestJpaEntity(HardMediumSoftBigDecimalScore score) {
-            this.score = score;
-        }
-
-        @Override
-        public HardMediumSoftBigDecimalScore getScore() {
-            return score;
-        }
-
-        @Override
-        public void setScore(HardMediumSoftBigDecimalScore score) {
-            this.score = score;
-        }
-
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoftlong/HardMediumSoftLongScoreHibernateTypeTest.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoftlong/HardMediumSoftLongScoreHibernateTypeTest.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoftlong/HardMediumSoftLongScoreHibernateTypeTest.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardmediumsoftlong/HardMediumSoftLongScoreHibernateTypeTest.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,52 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.hardmediumsoftlong;
-
-import javax.persistence.Column;
-import javax.persistence.Entity;
-
-import org.hibernate.annotations.Columns;
-import org.hibernate.annotations.TypeDef;
-import org.junit.jupiter.api.Test;
-import org.optaplanner.core.api.score.buildin.hardmediumsoftlong.HardMediumSoftLongScore;
-import org.optaplanner.persistence.jpa.impl.AbstractScoreJpaTest;
-
-import io.quarkus.test.junit.QuarkusTest;
-
-@QuarkusTest
-class HardMediumSoftLongScoreHibernateTypeTest extends AbstractScoreJpaTest {
-
-    @Test
-    void persistAndMerge() {
-        persistAndMerge(new HardMediumSoftLongScoreHibernateTypeTestJpaEntity(HardMediumSoftLongScore.ZERO),
-                HardMediumSoftLongScore.of(-100L, -20L, -3L),
-                HardMediumSoftLongScore.ofUninitialized(-7, -100L, -20L, -3L));
-    }
-
-    @Entity
-    @TypeDef(defaultForType = HardMediumSoftLongScore.class, typeClass = HardMediumSoftLongScoreHibernateType.class)
-    static class HardMediumSoftLongScoreHibernateTypeTestJpaEntity
-            extends AbstractTestJpaEntity<HardMediumSoftLongScore> {
-
-        @Columns(columns = { @Column(name = "initScore"),
-                @Column(name = "hardScore"), @Column(name = "mediumScore"), @Column(name = "softScore") })
-        protected HardMediumSoftLongScore score;
-
-        HardMediumSoftLongScoreHibernateTypeTestJpaEntity() {
-        }
-
-        public HardMediumSoftLongScoreHibernateTypeTestJpaEntity(HardMediumSoftLongScore score) {
-            this.score = score;
-        }
-
-        @Override
-        public HardMediumSoftLongScore getScore() {
-            return score;
-        }
-
-        @Override
-        public void setScore(HardMediumSoftLongScore score) {
-            this.score = score;
-        }
-
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoft/HardSoftScoreHibernateTypeTest.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoft/HardSoftScoreHibernateTypeTest.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoft/HardSoftScoreHibernateTypeTest.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoft/HardSoftScoreHibernateTypeTest.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,50 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.hardsoft;
-
-import javax.persistence.Column;
-import javax.persistence.Entity;
-
-import org.hibernate.annotations.Columns;
-import org.hibernate.annotations.TypeDef;
-import org.junit.jupiter.api.Test;
-import org.optaplanner.core.api.score.buildin.hardsoft.HardSoftScore;
-import org.optaplanner.persistence.jpa.impl.AbstractScoreJpaTest;
-
-import io.quarkus.test.junit.QuarkusTest;
-
-@QuarkusTest
-class HardSoftScoreHibernateTypeTest extends AbstractScoreJpaTest {
-
-    @Test
-    void persistAndMerge() {
-        persistAndMerge(new HardSoftScoreHibernateTypeTestJpaEntity(HardSoftScore.ZERO),
-                HardSoftScore.of(-10, -2),
-                HardSoftScore.ofUninitialized(-7, -10, -2));
-    }
-
-    @Entity
-    @TypeDef(defaultForType = HardSoftScore.class, typeClass = HardSoftScoreHibernateType.class)
-    static class HardSoftScoreHibernateTypeTestJpaEntity extends AbstractTestJpaEntity<HardSoftScore> {
-
-        @Columns(columns = { @Column(name = "initScore"), @Column(name = "hardScore"), @Column(name = "softScore") })
-        protected HardSoftScore score;
-
-        HardSoftScoreHibernateTypeTestJpaEntity() {
-        }
-
-        public HardSoftScoreHibernateTypeTestJpaEntity(HardSoftScore score) {
-            this.score = score;
-        }
-
-        @Override
-        public HardSoftScore getScore() {
-            return score;
-        }
-
-        @Override
-        public void setScore(HardSoftScore score) {
-            this.score = score;
-        }
-
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoftbigdecimal/HardSoftBigDecimalScoreHibernateTypeTest.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoftbigdecimal/HardSoftBigDecimalScoreHibernateTypeTest.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoftbigdecimal/HardSoftBigDecimalScoreHibernateTypeTest.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoftbigdecimal/HardSoftBigDecimalScoreHibernateTypeTest.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,56 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.hardsoftbigdecimal;
-
-import java.math.BigDecimal;
-
-import javax.persistence.Column;
-import javax.persistence.Entity;
-
-import org.hibernate.annotations.Columns;
-import org.hibernate.annotations.TypeDef;
-import org.junit.jupiter.api.Test;
-import org.optaplanner.core.api.score.buildin.hardsoftbigdecimal.HardSoftBigDecimalScore;
-import org.optaplanner.persistence.jpa.impl.AbstractScoreJpaTest;
-
-import io.quarkus.test.junit.QuarkusTest;
-
-@QuarkusTest
-class HardSoftBigDecimalScoreHibernateTypeTest extends AbstractScoreJpaTest {
-
-    @Test
-    void persistAndMerge() {
-        persistAndMerge(new HardSoftBigDecimalScoreHibernateTypeTestJpaEntity(HardSoftBigDecimalScore.ZERO),
-                HardSoftBigDecimalScore.of(new BigDecimal("-10.01000"), new BigDecimal("-2.20000")),
-                HardSoftBigDecimalScore.ofUninitialized(-7, new BigDecimal("-10.01000"), new BigDecimal("-2.20000")));
-    }
-
-    @Entity
-    @TypeDef(defaultForType = HardSoftBigDecimalScore.class, typeClass = HardSoftBigDecimalScoreHibernateType.class)
-    static class HardSoftBigDecimalScoreHibernateTypeTestJpaEntity
-            extends AbstractTestJpaEntity<HardSoftBigDecimalScore> {
-
-        @Columns(columns = {
-                @Column(name = "initScore"),
-                @Column(name = "hardScore", precision = 10, scale = 5),
-                @Column(name = "softScore", precision = 10, scale = 5) })
-        protected HardSoftBigDecimalScore score;
-
-        HardSoftBigDecimalScoreHibernateTypeTestJpaEntity() {
-        }
-
-        public HardSoftBigDecimalScoreHibernateTypeTestJpaEntity(HardSoftBigDecimalScore score) {
-            this.score = score;
-        }
-
-        @Override
-        public HardSoftBigDecimalScore getScore() {
-            return score;
-        }
-
-        @Override
-        public void setScore(HardSoftBigDecimalScore score) {
-            this.score = score;
-        }
-
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoftlong/HardSoftLongScoreHibernateTypeTest.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoftlong/HardSoftLongScoreHibernateTypeTest.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoftlong/HardSoftLongScoreHibernateTypeTest.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/hardsoftlong/HardSoftLongScoreHibernateTypeTest.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,50 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.hardsoftlong;
-
-import javax.persistence.Column;
-import javax.persistence.Entity;
-
-import org.hibernate.annotations.Columns;
-import org.hibernate.annotations.TypeDef;
-import org.junit.jupiter.api.Test;
-import org.optaplanner.core.api.score.buildin.hardsoftlong.HardSoftLongScore;
-import org.optaplanner.persistence.jpa.impl.AbstractScoreJpaTest;
-
-import io.quarkus.test.junit.QuarkusTest;
-
-@QuarkusTest
-class HardSoftLongScoreHibernateTypeTest extends AbstractScoreJpaTest {
-
-    @Test
-    void persistAndMerge() {
-        persistAndMerge(new HardSoftLongScoreHibernateTypeTestJpaEntity(HardSoftLongScore.ZERO),
-                HardSoftLongScore.of(-10L, -2L),
-                HardSoftLongScore.ofUninitialized(-7, -10L, -2L));
-    }
-
-    @Entity
-    @TypeDef(defaultForType = HardSoftLongScore.class, typeClass = HardSoftLongScoreHibernateType.class)
-    static class HardSoftLongScoreHibernateTypeTestJpaEntity extends AbstractTestJpaEntity<HardSoftLongScore> {
-
-        @Columns(columns = { @Column(name = "initScore"), @Column(name = "hardScore"), @Column(name = "softScore") })
-        protected HardSoftLongScore score;
-
-        HardSoftLongScoreHibernateTypeTestJpaEntity() {
-        }
-
-        public HardSoftLongScoreHibernateTypeTestJpaEntity(HardSoftLongScore score) {
-            this.score = score;
-        }
-
-        @Override
-        public HardSoftLongScore getScore() {
-            return score;
-        }
-
-        @Override
-        public void setScore(HardSoftLongScore score) {
-            this.score = score;
-        }
-
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/simple/SimpleScoreHibernateTypeTest.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/simple/SimpleScoreHibernateTypeTest.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/simple/SimpleScoreHibernateTypeTest.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/simple/SimpleScoreHibernateTypeTest.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,50 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.simple;
-
-import javax.persistence.Column;
-import javax.persistence.Entity;
-
-import org.hibernate.annotations.Columns;
-import org.hibernate.annotations.TypeDef;
-import org.junit.jupiter.api.Test;
-import org.optaplanner.core.api.score.buildin.simple.SimpleScore;
-import org.optaplanner.persistence.jpa.impl.AbstractScoreJpaTest;
-
-import io.quarkus.test.junit.QuarkusTest;
-
-@QuarkusTest
-class SimpleScoreHibernateTypeTest extends AbstractScoreJpaTest {
-
-    @Test
-    void persistAndMerge() {
-        persistAndMerge(new SimpleScoreHibernateTypeTestJpaEntity(SimpleScore.ZERO),
-                SimpleScore.of(-10),
-                SimpleScore.ofUninitialized(-7, -10));
-    }
-
-    @Entity
-    @TypeDef(defaultForType = SimpleScore.class, typeClass = SimpleScoreHibernateType.class)
-    static class SimpleScoreHibernateTypeTestJpaEntity extends AbstractTestJpaEntity<SimpleScore> {
-
-        @Columns(columns = { @Column(name = "initScore"), @Column(name = "score") })
-        protected SimpleScore score;
-
-        SimpleScoreHibernateTypeTestJpaEntity() {
-        }
-
-        public SimpleScoreHibernateTypeTestJpaEntity(SimpleScore score) {
-            this.score = score;
-        }
-
-        @Override
-        public SimpleScore getScore() {
-            return score;
-        }
-
-        @Override
-        public void setScore(SimpleScore score) {
-            this.score = score;
-        }
-
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/simplebigdecimal/SimpleBigDecimalScoreHibernateTypeTest.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/simplebigdecimal/SimpleBigDecimalScoreHibernateTypeTest.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/simplebigdecimal/SimpleBigDecimalScoreHibernateTypeTest.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/simplebigdecimal/SimpleBigDecimalScoreHibernateTypeTest.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,52 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.simplebigdecimal;
-
-import java.math.BigDecimal;
-
-import javax.persistence.Column;
-import javax.persistence.Entity;
-
-import org.hibernate.annotations.Columns;
-import org.hibernate.annotations.TypeDef;
-import org.junit.jupiter.api.Test;
-import org.optaplanner.core.api.score.buildin.simplebigdecimal.SimpleBigDecimalScore;
-import org.optaplanner.persistence.jpa.impl.AbstractScoreJpaTest;
-
-import io.quarkus.test.junit.QuarkusTest;
-
-@QuarkusTest
-class SimpleBigDecimalScoreHibernateTypeTest extends AbstractScoreJpaTest {
-
-    @Test
-    void persistAndMerge() {
-        persistAndMerge(new SimpleBigDecimalScoreHibernateTypeTestJpaEntity(SimpleBigDecimalScore.ZERO),
-                SimpleBigDecimalScore.of(new BigDecimal("-10.01000")),
-                SimpleBigDecimalScore.ofUninitialized(-7, new BigDecimal("-10.01000")));
-    }
-
-    @Entity
-    @TypeDef(defaultForType = SimpleBigDecimalScore.class, typeClass = SimpleBigDecimalScoreHibernateType.class)
-    static class SimpleBigDecimalScoreHibernateTypeTestJpaEntity extends AbstractTestJpaEntity<SimpleBigDecimalScore> {
-
-        @Columns(columns = { @Column(name = "initScore"), @Column(name = "score", precision = 10, scale = 5) })
-        protected SimpleBigDecimalScore score;
-
-        SimpleBigDecimalScoreHibernateTypeTestJpaEntity() {
-        }
-
-        public SimpleBigDecimalScoreHibernateTypeTestJpaEntity(SimpleBigDecimalScore score) {
-            this.score = score;
-        }
-
-        @Override
-        public SimpleBigDecimalScore getScore() {
-            return score;
-        }
-
-        @Override
-        public void setScore(SimpleBigDecimalScore score) {
-            this.score = score;
-        }
-
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/simplelong/SimpleLongScoreHibernateTypeTest.java
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/simplelong/SimpleLongScoreHibernateTypeTest.java b/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/simplelong/SimpleLongScoreHibernateTypeTest.java
deleted file mode 100644
--- a/optaplanner-persistence/optaplanner-persistence-jpa/src/test/java/org/optaplanner/persistence/jpa/impl/score/buildin/simplelong/SimpleLongScoreHibernateTypeTest.java	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ /dev/null	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
@@ -1,50 +0,0 @@
-package org.optaplanner.persistence.jpa.impl.score.buildin.simplelong;
-
-import javax.persistence.Column;
-import javax.persistence.Entity;
-
-import org.hibernate.annotations.Columns;
-import org.hibernate.annotations.TypeDef;
-import org.junit.jupiter.api.Test;
-import org.optaplanner.core.api.score.buildin.simplelong.SimpleLongScore;
-import org.optaplanner.persistence.jpa.impl.AbstractScoreJpaTest;
-
-import io.quarkus.test.junit.QuarkusTest;
-
-@QuarkusTest
-class SimpleLongScoreHibernateTypeTest extends AbstractScoreJpaTest {
-
-    @Test
-    void persistAndMerge() {
-        persistAndMerge(new SimpleLongScoreHibernateTypeTestJpaEntity(SimpleLongScore.ZERO),
-                SimpleLongScore.of(-10L),
-                SimpleLongScore.ofUninitialized(-7, -10L));
-    }
-
-    @Entity
-    @TypeDef(defaultForType = SimpleLongScore.class, typeClass = SimpleLongScoreHibernateType.class)
-    static class SimpleLongScoreHibernateTypeTestJpaEntity extends AbstractTestJpaEntity<SimpleLongScore> {
-
-        @Columns(columns = { @Column(name = "initScore"), @Column(name = "score") })
-        protected SimpleLongScore score;
-
-        SimpleLongScoreHibernateTypeTestJpaEntity() {
-        }
-
-        public SimpleLongScoreHibernateTypeTestJpaEntity(SimpleLongScore score) {
-            this.score = score;
-        }
-
-        @Override
-        public SimpleLongScore getScore() {
-            return score;
-        }
-
-        @Override
-        public void setScore(SimpleLongScore score) {
-            this.score = score;
-        }
-
-    }
-
-}
Index: optaplanner-persistence/optaplanner-persistence-jpa/pom.xml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/optaplanner-persistence/optaplanner-persistence-jpa/pom.xml b/optaplanner-persistence/optaplanner-persistence-jpa/pom.xml
--- a/optaplanner-persistence/optaplanner-persistence-jpa/pom.xml	(revision 304972c91ca5b3c5db7addc3527564cb5d28fdba)
+++ b/optaplanner-persistence/optaplanner-persistence-jpa/pom.xml	(date 1679065600012)
@@ -43,30 +43,6 @@
       <groupId>jakarta.persistence</groupId>
       <artifactId>jakarta.persistence-api</artifactId>
     </dependency>
-    <!-- Hibernate -->
-    <dependency>
-      <groupId>org.hibernate</groupId>
-      <artifactId>hibernate-core</artifactId>
-      <exclusions>
-        <!-- Duplicates classes of jakarta.activation:jakarta.activation-api -->
-        <exclusion>
-          <groupId>javax.activation</groupId>
-          <artifactId>javax.activation-api</artifactId>
-        </exclusion>
-        <!-- Duplicates classes of jakarta.xml.bind:jakarta.xml.bind-api -->
-        <exclusion>
-          <groupId>javax.xml.bind</groupId>
-          <artifactId>jaxb-api</artifactId>
-        </exclusion>
-        <!-- TODO: remove once the hibernate-core switches to jakarta.transaction:jakarta.transaction-api. -->
-        <exclusion>
-          <groupId>org.jboss.spec.javax.transaction</groupId>
-          <artifactId>jboss-transaction-api_1.2_spec</artifactId>
-        </exclusion>
-      </exclusions>
-      <!-- Used at compile time by *ScoreHibernateType -->
-      <optional>true</optional>
-    </dependency>
 
     <!-- Testing -->
     <dependency>
